odoo/addons/base/models/res_company.py

```diff
@@ -248,19 +252,16 @@ def open_company_edit_report(self):
         return self.env['res.config.settings'].open_company()
 
     @api.multi
-    def write_company_and_print_report(self, values):
-        res = self.write(values)
-
-        report_name = values.get('default_report_name')
-        active_ids = values.get('active_ids')
-        active_model = values.get('active_model')
+    def write_company_and_print_report(self):
+        context = self.env.context
+        report_name = context.get('default_report_name')
+        active_ids = context.get('active_ids')
+        active_model = context.get('active_model')
         if report_name and active_ids and active_model:
             docids = self.env[active_model].browse(active_ids)
             return (self.env['ir.actions.report'].search([('report_name', '=', report_name)], limit=1)
-                        .with_context(values)
+                        .with_context(context)
```

Comment:
By default `X.search()` returns a recordset with the same `env` as `X`.
In this case, the output has `self.env` as environment.
Therefore `.with_context(self.env.context)` is useless...